src/sparsebase/preprocess/preprocess.h

```diff
@@ -826,6 +826,120 @@ class Degrees_DegreeDistribution
   void Register();
 };
 
+//! An abstract class representing partitioning algorithms.
+/*!
+ * Class that generalizes partitioning algorithms. It defines the API used for
+ * partitioning as well as the return type of partitioning (IDType*).
+ * @tparam IDType  the data type of row and column numbers (vertex IDs in the
+ * case of graphs)
+ */
+template <typename IDType>
+class PartitionPreprocessType : public FunctionMatcherMixin<IDType *> {
+public:
+  PartitionPreprocessType();
+
+  //! Performs a partition operation using the default parameters
+  /*!
+   * @returns An IDType array where the i-th index contains the ID for the partitioning i belongs to
+   */
+  IDType* Partition(format::Format * format, std::vector<context::Context*> contexts, bool convert_input);
+
+  //! Performs a partition operation using the parameters supplied by the user
+  /*!
+   * @returns An IDType array where the i-th index contains the ID for the partitioning i belongs to
+   */
+  IDType* Partition(format::Format *format, PreprocessParams *params,
+                     std::vector<context::Context *> contexts, bool convert_input);
+  virtual ~PartitionPreprocessType();
+};
+
+
+#ifdef USE_METIS
+
+
+//! A wrapper for the METIS partitioner
+/* !
+ * Wraps the METIS partitioner available here: https://github.com/KarypisLab/METIS
+ * The library must be compiled with the USE_METIS option turned on
+ * and the pre-built METIS library should be available.
+ * See the Optional Dependencies page (under Getting Started) in our documentation for more info.
```

Comment:
It seems there aren't any updates to the getting started page in this PR. will this be added later or did you forget to add the file 😂

Reply:
I have another branch with that file added. But I think I didn't intend to write this reference before that merged. For some reason I forgot that and added it in. I will make another PR soon with the updated docs.